src/components/newNote/NewNote.jsx

```diff
@@ -1,43 +1,111 @@
 import "./notes.css";
-import { BiPin, BiPalette } from "assets/icons/Icons";
+import { BiPin, BiPinFill } from "assets/icons/Icons";
+import { useState } from "react";
+import { ColorPalette } from "components/index";
+import { useNotes } from "context";
 
 function NewNote() {
+  const [formDetails, setFormDetails] = useState({
+    title: "",
+    description: "",
+    label: "",
+    priority: "",
+    color: "#fff",
+    isPinned: false,
+  });
+  const { addNewNoteHandler } = useNotes();
+
   return (
-    <div className="input-notes-container">
-      <button className="btn-pin" type="button">
-        <BiPin className="fs-m" />
-      </button>
+    <div
+      className="input-notes-container"
+      style={{ backgroundColor: formDetails.color }}
+    >
+      {formDetails.isPinned ? (
+        <button
+          className="btn-pin"
+          onClick={() => {
+            setFormDetails({ ...formDetails, isPinned: !formDetails.isPinned });
+          }}
+        >
+          <BiPinFill className="fs-m" />
```

Comment:
The `button` tag is common. Since you are only changing the icon, you can move it outside the `formDetails.isPinned ?` 
and check if pinned for Icons directly.

`<button... 
{formDetails.isPinned ? <BiPinFill className="fs-m" /> ......}
</..>'

Reply:
Done!